go/host/storage/hostdb/hostdb.go

```diff
@@ -0,0 +1,75 @@
+package hostdb
+
+import (
+	"database/sql"
+	"fmt"
+)
+
+type HostDB interface {
+	GetSQLDB() *sql.DB
+	NewDBTransaction() *dbTransaction
+	BeginTx() (*sql.Tx, error)
+	GetSQLStatement() *SQLStatements
+}
+
+type hostDB struct {
+	sqldb      *sql.DB
+	statements *SQLStatements
+}
+
+func (db *hostDB) GetSQLStatement() *SQLStatements {
+	return db.statements
+}
+
+func NewHostDB(db *sql.DB, statements *SQLStatements) (HostDB, error) {
+	return &hostDB{
+		sqldb:      db,
+		statements: statements,
+	}, nil
+}
+
+func (db *hostDB) GetSQLDB() *sql.DB {
+	return db.sqldb
+}
+
+func (db *hostDB) BeginTx() (*sql.Tx, error) {
+	return db.sqldb.Begin()
+}
+
+func (db *hostDB) NewDBTransaction() *dbTransaction {
```

Comment:
I don't think this will work.
The dbTransaction wrapper should be on top of a sql.Tx.

Reply:
the `BeginTx` method is on top of the `sql.Tx` which is called in the `Write` method [here](https://github.com/ten-protocol/go-ten/blob/2ba4b62afbb840d413cf2740a5a25ece625beb97/go/host/storage/hostdb/hostdb.go#L65).

I ran the full network tests over the deployed db yesterday and it did pass.

Reply:
Understood - switched it out!